app/components/swipeable.js

```diff
@@ -0,0 +1,83 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class SwipeableComponent extends Component {
+  touching = false
+  swiping = false
+  xStart = null
+  yStart = null
+  xEnd = null
+  yEnd = null
+  @tracked
+  swipeDirection = this.args.onSwipeHorizontal ? 'horizontal' : (this.args.onSwipeVertical ? 'vertical' : null)
+
+  reset() {
+    this.touching = false;
+    this.swiping = false;
+    this.xStart = null;
+    this.yStart = null;
+    this.xEnd = null;
+    this.yEnd = null;
+  }
+
+  @action
+  onTouchStart(touchStartEvent) {
+    if (touchStartEvent.targetTouches.length === 1) {
+      this.touching = true;
+      let touch = touchStartEvent.targetTouches.item(0);
```

Comment:
Can you use `const` instead of `let` on places where the variables do not change?

Reply:
Sure, but shouldn't small things like these be caught by the linter?